### ISSUE TITLE
Reduce training time？

### ISSUE THREAD

**Issue by zhoumengbo** [Dao-AILab/flash-attention#642]
Hello author, I noticed in the paper that when training a GPT-style model, the speed of the second generation improved by 1.3 times. I'm curious, does this imply that if I pre-train a model from scratch, with all other conditions remaining the same, using FlashAttention2 can reduce the overall training time by 1.3 times compared to using FlashAttention?

**Comment by tridao** [comment]
Yes, given the same setting as in the paper (e.g. context length 8k, 3B model, 8xA100). Ofc for other settings (model size, context length, hardware) the speedup will be different.

**Comment by zhoumengbo** [comment]
> FlashAttention2

I want to know what percentage impact you think the Attention has on the total pre-training time. If I assume the impact of the Attention is 50%, then the remaining factors also account for 50%. If the time consumed by FlashAttention2 is x% FlashAttention, then 50% * x% + 50% * 1 = 1/1.3, from which we can derive x% = 53.85%. This is roughly in line with the approximately 2x speedup mentioned in the paper. Therefore, my assumption holds. Do you think this line of reasoning is valid?

**Comment by zhoumengbo** [comment]
> Yes, given the same setting as in the paper (e.g. context length 8k, 3B model, 8xA100). Ofc for other settings (model size, context length, hardware) the speedup will be different.

I want to know what percentage impact you think the Attention has on the total pre-training time. If I assume the impact of the Attention is 50%, then the remaining factors also account for 50%. If the time consumed by FlashAttention2 is x% FlashAttention, then 50% * x% + 50% * 1 = 1/1.3, from which we can derive x% = 53.85%. This is roughly in line with the approximately 2x speedup mentioned in the paper. Therefore, my assumption holds. Do you think this line of reasoning is valid?
